### PR TITLE
Wagtail 3 support

### PR DESCRIPTION
**Pull request by nickmoreton**
This PR builds on https://github.com/wagtail/wagtail-personalisation/pull/239 to add support for Wagtail v3+

It replaces https://github.com/wagtail/wagtail-personalisation/pull/236

### CODE REVIEW

File: tests/unit/test_static_dynamic_segments.py [hunk 1]
```diff
@@ -21,10 +21,22 @@ def form_with_data(segment, *rules):
         "randomisation_percent",
     ]
 
+    model_formsets = [
```
Comment (nickmoreton) [inline tests/unit/test_static_dynamic_segments.py:24]: **Comment** https://github.com/wagtail/django-modelcluster v6.0 now needs the `formsets` to be declared

File: src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<li class="icon icon-fa-snowflake-o">
```
Comment (William-Blackie) [inline src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html:1]: Question: Why is this new?

File: tests/settings.py [hunk 3]
```diff
@@ -1,98 +1,88 @@
...
-    'tests.site.pages',
+    "wagtail_personalisation",
+    "wagtail.contrib.modeladmin",
+    "wagtail.search",
+    "wagtail.sites",
+    "wagtail.users",
+    "wagtail.images",
+    "wagtail.documents",
+    "wagtail.admin",
+    "wagtail" if WAGTAIL_VERSION >= (3, 0) else "wagtail.core",
```
Comment (William-Blackie) [inline tests/settings.py:73]: Praise: Nice

File: tests/unit/test_models.py [hunk 4]
```diff
@@ -76,8 +78,8 @@ def test_sitemap_generation_for_variants_is_disabled(segmented_page):
 @pytest.mark.django_db
 def test_segment_edit_view(site, client, django_user_model):
     test_segment = SegmentFactory()
-    try:
-        new_panel = test_segment.panels[1].children[0].bind_to(model=Segment)
-    except AttributeError:
```
Comment (William-Blackie) [inline tests/unit/test_models.py:81]: Question: What's the attribute error doing originally? Why do we not need it now?

File: src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html (hunk 2 above)
Comment (nickmoreton) [reply src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html:1]: SiteSummaryPanel is a subclass of Component. The Component class no longer has a `render` method. It's actually a change first implemented in v2.15: https://docs.wagtail.org/en/stable/releases/2.15.html#admin-homepage-panels-summary-items-and-action-menu-items-now-use-components

This is where the change was made: https://github.com/wagtail/wagtail-personalisation/pull/240/commits/241af6cd57bfe46e8edadaf225db778338f73e1a#diff-c4a6309c46bb6f60b100be17b7b319c4e96fe47ff4d849ba095e39f25e06b3eaL249 and the template was added for that reason. Should have been in the same commit 😭 sorry.

Do let me know if I've described it right 😄 Thanks

File: tests/unit/test_models.py (hunk 4 above)
Comment (nickmoreton) [reply tests/unit/test_models.py:81]: I'm not entirely sure why that try/except was there other than to check for presence of the  `bind_to_model` method but I expect is was a change for an earlier Wagtail version possibly here: https://docs.wagtail.org/en/v2.13.5/releases/2.5.html#upgrade-considerations

As we are supporting Wagtail v2.9+ now we probably don't even need this version check. What do you think @William-Blackie

File: src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html (hunk 2 above)
Comment (William-Blackie) [reply src/wagtail_personalisation/templates/modeladmin/wagtail_personalisation/segment/summary.html:1]: Ah okay. Makes more sense.

File: tests/unit/test_models.py (hunk 4 above)
Comment (William-Blackie) [reply tests/unit/test_models.py:81]: Ah right, that makes sense. Thanks for explaining @nickmoreton

**Pull request closed without merging**